packages/editor/src/components/parts/common/responsible/useRoles.ts

```diff
@@ -5,22 +5,29 @@ import { EMPTY_ROLE, type RoleMeta } from '@axonivy/inscription-protocol';
 
 export const useRoles = (showTaskRoles = false) => {
   const [roles, setRoles] = useState<SelectItem[]>([]);
+  const [rolesAsTree, setRolesAsTree] = useState<RoleMeta[]>([]);
   const { context, elementContext } = useEditorContext();
   const roleTree = useMeta('meta/workflow/roleTree', context, EMPTY_ROLE).data;
   const taskRoles = useMeta('meta/workflow/taskRoles', elementContext, []).data;
 
   useEffect(() => {
     const flatRoles: SelectItem[] = [];
-    const addRoles = (role: RoleMeta) => {
+    const treeRoles: RoleMeta[] = [];
+    const addFlatRoles = (role: RoleMeta) => {
       flatRoles.push({ label: role.id, value: role.id });
-      role.children.forEach(addRoles);
+      role.children.forEach(addFlatRoles);
+    };
+    const addTreeRoles = (role: RoleMeta) => {
+      treeRoles.push(role);
     };
     if (showTaskRoles) {
-      taskRoles.forEach(addRoles);
+      taskRoles.forEach(role => (addTreeRoles(role), addFlatRoles(role)));
```

Comment:
even if this is maybe correct code it looks strange to me... I would write a more clear block for it, and IMO there is not really a need to create this addTreeRoles method: 
```ts
taskRoles.forEach(role => {
  treeRoles.push(role);
  addFlatRoles(role)
});
```